sw/device/lib/dif/dif_clkmgr.h

```diff
@@ -0,0 +1,200 @@
+ * ignore the request altogether.
+ */
+typedef uint32_t dif_clkmgr_hintable_clock_t;
+
+/**
+ * Creates a new handle for clock manager.
+ *
+ * This function does not actuate the hardware.
+ *
+ * @param params Hardware instantiation parameters.
+ * @param[out] handle Out param for the initialized handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_init(dif_clkmgr_params_t params,
+                                    dif_clkmgr_t *handle);
+
+/**
+ * Check if a Gateable Clock is Enabled or Disabled.
+ *
+ * @param handle Clock Manager Handle.
+ * @param clock Gateable Clock ID.
+ * @param[out] is_enabled whether the clock is enabled or not.
+ * @returns The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_gateable_clock_is_enabled(
+    const dif_clkmgr_t *handle, dif_clkmgr_gateable_clock_t clock,
+    bool *is_enabled);
+
+/**
+ * Enable or Disable a Gateable Clock.
+ *
+ * @param handle Clock Manager Handle.
+ * @param clock Gateable Clock ID.
+ * @param new_state whether to enable or disable the clock.
+ * @returns The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_gateable_clock_toggle(
+    const dif_clkmgr_t *handle, dif_clkmgr_gateable_clock_t clock,
+    dif_clkmgr_toggle_t new_state);
+
+/**
+ * Check if a Hintable Clock is Enabled or Disabled.
+ *
+ * Hintable clocks are not under full software control, but this operation
+ * accurately reflects the state of the current clock, regardless of any
+ * requests made by software.
+ *
+ * @param handle Clock Manager Handle.
+ * @param clock Hintable Clock ID.
+ * @param[out] is_enabled whether the clock is enabled or not.
+ * @returns The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_hintable_clock_is_enabled(
+    const dif_clkmgr_t *handle, dif_clkmgr_hintable_clock_t clock,
+    bool *is_enabled);
+
+/**
+ * Enable or Disable a Hintable Clock.
+ *
+ * This only sets a hint that the clock should be enabled or disabled. The clock
+ * manager is in control of whether the clock should actually be enabled or
+ * disabled.
+ *
+ * To read what hint the software has given the hardware, use
+ * #dif_clkmgr_hintable_clock_read_hint. To read whether the clock is currently
+ * enabled or disabled, use #dif_clkmgr_hintable_clock_is_enabled.
+ *
+ * @param handle Clock Manager Handle.
+ * @param clock Hintable Clock ID.
+ * @param new_state whether to enable or disable the clock.
+ * @returns The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_hintable_clock_toggle(
+    const dif_clkmgr_t *handle, dif_clkmgr_hintable_clock_t clock,
+    dif_clkmgr_toggle_t new_state);
+
+/**
+ * Read the current Hint for a Hintable Clock.
+ *
+ * Hintable clocks are not under full software control, but this operation
+ * accurately reflects the state of the current clock, regardless of any
```

Comment:
This explanation I find confusing. This function allows you to find out what hint is set for the clock, but `accurately reflects the state of the current clock` but this is almost contradictory.

Reply:
Am I understanding correctly that this function is actually reading `CLK_HINTS_STATUS`, which gives the "real" state of the clock? I think it should be named `get_status` or `get_state` to avoid confusion.

Reply:
We agree with @silvestrst, if the function reads the actual state of the clock then the name should reflect that it's reading the state not the hint. This looks like it's maybe just a copy and paste from the description above, so perhaps unintentional?

Reply:
Yeah, this was a copy/paste that was not updated. New naming and description should be clearer.